cpu/cc2538/dev/spi.c

```diff
@@ -41,102 +41,337 @@
 #include "dev/spi.h"
 #include "dev/ssi.h"
 #include "dev/gpio.h"
+/*---------------------------------------------------------------------------*/
+/* Map the old port / pin names to the new ones for compatibility reasons */
+#if (SPI_DEFAULT_INSTANCE == 0)
+#ifdef SPI_CLK_PORT
+#define SPI0_CLK_PORT         SPI_CLK_PORT
+#endif
+#ifdef SPI_CLK_PIN
+#define SPI0_CLK_PIN          SPI_CLK_PIN
+#endif
+#ifdef SPI_MOSI_PORT
+#define SPI0_TX_PORT          SPI_MOSI_PORT
+#endif
+#ifdef SPI_MOSI_PIN
+#define SPI0_TX_PIN           SPI_MOSI_PIN
+#endif
+#ifdef SPI_MISO_PORT
+#define SPI0_RX_PORT          SPI_MISO_PORT
+#endif
+#ifdef SPI_MISO_PIN
+#define SPI0_RX_PIN           SPI_MISO_PIN
+#endif
+#elif (SPI_DEFAULT_INSTANCE == 1)
+#ifdef SPI_CLK_PORT
+#define SPI1_CLK_PORT         SPI_CLK_PORT
+#endif
+#ifdef SPI_CLK_PIN
+#define SPI1_CLK_PIN          SPI_CLK_PIN
+#endif
+#ifdef SPI_MOSI_PORT
+#define SPI1_TX_PORT          SPI_MOSI_PORT
+#endif
+#ifdef SPI_MOSI_PIN
+#define SPI1_TX_PIN           SPI_MOSI_PIN
+#endif
+#ifdef SPI_MISO_PORT
+#define SPI1_RX_PORT          SPI_MISO_PORT
+#endif
+#ifdef SPI_MISO_PIN
+#define SPI1_RX_PIN           SPI_MISO_PIN
+#endif
+#else
+#error "Invalid SPI instance. Valid values are 0 or 1"
+#endif
```

Comment:
The legacy SPI_* definitions above would only come from board header files. They are not part of the core SPI API. Hence, we don't care about them. They can just be removed, and out-of-tree users should just slightly upgrade their code. Then, board.h files should directly define SPI pins as SPI0_* or SPI1_*.